### ISSUE TITLE
奇怪的精度测量方式

### ISSUE THREAD

**Issue by wds1996** [SUNSTARK/MCLP#1]
在你的代码中有个很奇怪的设定。    
with torch.no_grad():
        for batch_data in dataloader:
            location_output = model(batch_data)
            location_y = batch_data['location_y']
            location_y = location_y.view(-1)
            total_samples += location_y.size(0)

            top_k_correct_loc += evaluate(location_output, location_y, top_k_values)
            precision_loc += calculate_mrr(location_output, location_y)
上述是test时候的精度计算，据我所知，大家都是只计算最后预测结果的精度，但是在你的代码里把整个序列都算进去了。
奇怪的是，你使用transformerEncoder，这是不带掩码的，我目前也没发现掩码传入，这就导致模型完全可以把前面的答案记住。

一个简单的测试，如果使用上述的精度测量方式，一个简单的transformer的encoder和decoder架构随便都能达到90%的精度。

**Comment by SUNSTARK** [comment]
感谢你的关注。

我们在MCLP中仅使用了Transformer的Encoder部分。为了避免信息泄露（future information leakage），在输入时应用了未来位置掩码（mask），对应代码如下（位于framework.py的56-57行）:
  ```python
future_mask = torch.triu(torch.ones(sequence_length, sequence_length), diagonal=1).to(lt_embedded.device)
future_mask = future_mask.masked_fill(future_mask == 1, float('-inf')).bool()
```

在训练与测试阶段，我们的设定是：对于给定的一条完整序列，模型需要在每一个时间步预测下一个位置。
这种全序列预测的设定，与Flashback、Graph-Flashback、SNPM等已有工作保持一致，因此在评估时也需要基于整个序列计算预测指标，而不是仅计算最终一个位置的预测精度。
